Enable CuDNN v8 frontend in RL

malfet opened the pull request:
Summary:
This enables use of CUDNN v8 in all Meta internal workflows. Also, fixes two minor issues:
- Skip LogCumSumExp compilation for complex dtypes for fbcode and RL
- Move `MakeConvOutputShape` template definition/specialization to anonymous namespace inside `at::native::quantized` as it is referenced from both `torch_cpu` and `torch_cuda`. This is necessary to avoid `duplicate symbol` linker error if say `libtorch_cpu` and `libtorch_cuda` are statically linked together.
- Lower CuDNN v8 version guard from 8.3 to 8.2 (as there are no good reason why it should be 8.3, first version of the library that properly supports all the features is actually 8.5)

Test Plan: CI

Differential Revision: D46161651



cc @seemethere @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

vadimkantorov commented on aten/src/ATen/native/quantized/ConvUtils.cpp:
```diff
@@ -0,0 +1,47 @@
+#if defined(USE_CUDA) || defined(USE_PYTORCH_QNNPACK)
+namespace at::native::quantized {
+template <>
+at::SmallVector<int64_t, 4> MakeConvOutputShape<2>(
```
are these related to quantized convs specifically? if not, maybe worth moving them up in the namespace?

malfet replied:
They are only used in quantized conv, though there is nothing that screams quantized here. But I'll likely have to move them to anonymous namespace and force inlining as Windows linking quirks drive me crazy.

vadimkantorov replied:
The reason I'm asking is that in more frontend context it was decided that this formula should not be available explicitly and that meta tensors + module/function call should be done for computing this :) So I thought that this might weigh on the other side and a simple user-facing formula can be provided in some torch.nn.utils :)

malfet replied:
Makes sense, and I guess it could.

malfet committed:
Enable CuDNN v8 frontend in RL (#102284)

Summary:
This enables use of CUDNN v8 in all Meta internal workflows. Also, fixes two minor issues:
- Skip LogCumSumExp compilation for complex dtypes for fbcode and RL
- Move `MakeConvOutputShape` template definition/specialization to anonymous namespace inside `at::native::quantized` as it is referenced from both `torch_cpu` and `torch_cuda`. This is necessary to avoid `duplicate symbol` linker error if say `libtorch_cpu` and `libtorch_cuda` are statically linked together.
- Lower CuDNN v8 version guard from 8.3 to 8.2 (as there are no good reason why it should be 8.3, first version of the library that properly supports all the features is actually 8.5)

Pull Request resolved: https://github.com/pytorch/pytorch/pull/102284

Test Plan: buck2 build arvr/mode/platform010/cuda11/opt //xplat/caffe2:_C

Reviewed By: atalman

Differential Revision: D46161651

Pulled By: malfet

fbshipit-source-id: 796a11951efcef6da7901806f091fc3927ee4867
